applications/IgaApplication/custom_conditions/support_laplacian_condition.cpp

```diff
@@ -0,0 +1,157 @@
+                    DN_dot_n_vec(i)  += DN_DX(i, idim) * normal_physical_space[idim];
+                } 
+                
+            }
+            // Differential area
+            double penalty_integration = penalty * integration_points[point_number].Weight(); //  * std::abs(DetJ0);
+
+            // Collins, Lozinsky & Scovazzi innovation
+            double Guglielmo_innovation = 1.0;  // = 1 -> Penalty approach
+                                                // = -1 -> Free-penalty approach
+            if (penalty == -1.0) {
+                penalty_integration = 0.0;
+                Guglielmo_innovation = -1.0;
+            }
+            
+
+            // Assembly
+            noalias(rLeftHandSideMatrix) -= prod(trans(H), H) * penalty_integration;
+            // Assembly of the integration by parts term -(w,GRAD_u * n) -> Fundamental !!
+            noalias(rLeftHandSideMatrix) -= prod(trans(H), DN_dot_n)                        * integration_points[point_number].Weight(); // * std::abs(DetJ0) ;
+            // Assembly Dirichlet BCs -(GRAD_w* n,u) 
+            noalias(rLeftHandSideMatrix) -= Guglielmo_innovation * prod(trans(DN_dot_n), H) * integration_points[point_number].Weight(); // * std::abs(DetJ0) ;
+
+
+            if (CalculateResidualVectorFlag) {
+                
+                const double u_D_scalar = this->GetValue(TEMPERATURE);
+
+                noalias(rRightHandSideVector) -=  H_vector * u_D_scalar * penalty_integration;
+                noalias(rRightHandSideVector) -= Guglielmo_innovation * DN_dot_n_vec * u_D_scalar * integration_points[point_number].Weight(); // * std::abs(DetJ0);
+
+                Vector temp(number_of_nodes);
+                // RHS = ExtForces - K*temp;
+                for (IndexType i = 0; i < number_of_nodes; i++) {
+                    temp[i] = r_geometry[i].GetSolutionStepValue(TEMPERATURE);
+                }
+                // RHS -= K*temp
+                noalias(rRightHandSideVector) -= prod(rLeftHandSideMatrix,temp);
+
+            }
+        }
+        KRATOS_CATCH("")
+    }
+
+    int SupportLaplacianCondition::Check(const ProcessInfo& rCurrentProcessInfo) const
+    {
+        KRATOS_ERROR_IF_NOT(GetProperties().Has(PENALTY_FACTOR))
+            << "No penalty factor (PENALTY_FACTOR) defined in property of SupportLaplacianCondition" << std::endl;
+        return 0;
+    }
+
+    void SupportLaplacianCondition::EquationIdVector(
+        EquationIdVectorType& rResult,
+        const ProcessInfo& rCurrentProcessInfo
+    ) const
+    {
+        const auto& r_geometry = GetGeometry();
+        const SizeType number_of_nodes = r_geometry.size();
+
+        if (rResult.size() !=  number_of_nodes)
+            rResult.resize(number_of_nodes, false);
+
+        for (IndexType i = 0; i < number_of_nodes; ++i) {
+            const auto& r_node = r_geometry[i];
+            rResult[i] = r_node.GetDof(TEMPERATURE).EquationId();

```

Comment:
Shouldn't you take this as the unknown variable from the convection diffusion settings?